### ISSUE TITLE
[v4] Publish @tailwindcss/postcss as ESM

### ISSUE THREAD

**Issue by wesbos** [tailwindlabs/tailwindcss#13692]
"@tailwindcss/postcss": "^4.0.0-alpha.15" is currently published as CJS,  while "@tailwindcss/vite": "^4.0.0-alpha.15" is published as ESM.

Trying to use the postcss plugin inside a vite config file like this (with type: module set in package.json)

```js
import tailwindcssPostCSS from '@tailwindcss/postcss'
import { defineConfig } from 'vite'
export default defineConfig({
  css: {
    postcss: {
      plugins: [tailwindcssPostCSS]
    },
  },
})

```

Causes this issue:

✘ [ERROR] Failed to resolve entry for package "@tailwindcss/postcss". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." specifier in "@tailwindcss/postcss" package [plugin externalize-deps]

    node_modules/esbuild/lib/main.js:1374:27:
      1374 │         let result = await callback({
           ╵                            ^

a workaround currently is to use an external postcss.config.js file:

```js
export default {
  plugins: {
    "@tailwindcss/postcss": {},
  },
};
```

**Comment by wesbos** [comment]
Ah - the exports are incorrect. This package is fine as CJS, as long as we fix the exports. 

https://github.com/tailwindlabs/tailwindcss/pull/13693

**Comment by RobinMalfait** [comment]
Hey!

This should be fixed by #13693, and will be available in the next release.
